### DISCUSSION TITLE
Running Linux Gtk App by Windows WSL

### DISCUSSION THREAD

**Discussion by maroonedsia** [unoplatform/uno#14680] [Q&A] [answered]
Hi, following the instructions on [Get Started on Visual Studio 2022](https://platform.uno/docs/articles/get-started-vs-2022.html) and and [Using the Skia+GTK head](https://platform.uno/docs/articles/features/using-skia-gtk.html), I could not run the default `UnoDemo.Skia.Gtk` app through Visual Studio on WSL. The error was:

```
Unable to initialize Gtk, visit https://aka.platform.uno/gtk-install for more information.
The program 'dotnet' has exited with code 0 (0x0).
```

Then, I figured out I need to run these commands on my WSL's Ubuntu (mine is 22.04). I've lost the source, but this actually worked for me:

```bash
sudo apt update

# - Ubuntu 18.04 => 
sudo apt-get install gtk+3.0 mesa-utils libgl1-mesa-glx ttf-mscorefonts-installer

# - Ubuntu 20.04/22.04 => 
sudo apt install libgtk-3-dev mesa-utils libgl1-mesa-glx ttf-mscorefonts-installer
```

So, could an expert explain the situation to the rest of us?

A. If this is required, why it's not a part of the official docs?
B. Is installing [GTK3 runtime](https://github.com/tschoonj/GTK-for-Windows-Runtime-Environment-Installer/releases) on Windows still required, or updating the Ubuntu's graphic library, as above, suffices?

**Comment by jeromelaban** [comment, marked as answer]
Thanks for the report. 

> A. If this is required, why it's not a part of the official docs?

Indeed, the documentation does not state to install GTK for WSL, though [it's mentioned here](https://platform.uno/docs/articles/get-started-vscode.html?tabs=linux%2Cubuntu1804#platform-specific-setup). The documented is not up to date in the https://aka.platform.uno/gtk-install, thanks for point this out.

> B. Is installing [GTK3 runtime](https://github.com/tschoonj/GTK-for-Windows-Runtime-Environment-Installer/releases) on Windows still required, or updating the Ubuntu's graphic library, as above, suffices?

If you're running directly on windows, you'll need the GTK3 runtime for windows. If you're running with WSL, you will not need the windows GTK3 runtime for windows, only the linux packages.

**Reply by maroonedsia** [reply]
Got it, and thanks for the confirmation. 👍
